src/Elastic.Apm/Model/Payload/Transaction.cs

```diff
@@ -51,6 +53,21 @@ public Transaction(IApmLogger logger, string name, string type, IPayloadSender s
 
 		public string Name { get; set; }
 
+		public Api.Request Request
+		{
+			get => _request;
+			set
+			{
+				_request = value;
+				_context.Value.Request = new Request(value.Method,
```

Comment:
This can be also done during serialization... I'm not sure where to do this... I'm ok with moving this.

Reply:
🤦‍♂️  We have to... this won't work if we do it here.

Reply:
Done, moved this to `End()`